doc/contributing/result-object-and-error-handling.md

```diff
@@ -0,0 +1,77 @@
+# Result object and Error Handling
+
+
+## Result object
+
+The Result class is similar to Option type/object in Scala/Rust.
+It wraps up some `value` and `error` - so that a function can return both simulateously.
+
+This is similar to the https://www.npmjs.com/package/neverthrow package
+
+### Simple usage
+
+```ts
+function someFn(someArg): Result<string | undefined> {
+  if (someArg) {
+    return new Result('success')
+  }
+  return new Result(undefined, TerriaError.from("SOME ERROR"))
+}
+```
+
+#### This can now be used in a few different ways
+
+##### Ignore error and just return value
+
+```ts
+const value = someFn(someArg).ignoreError()
+```
+
+##### Catch error, do something with it and then return value
+
+```ts
+const value = someFn(someArg).catchError((error) => doSomethingWithError(error))
+```
+
+##### Throw error OR return value
+
+```ts
+const value = someFn(someArg).throwIfError()
+```
+
+##### Throw if value is undefined, otherwise return value
+
+```ts
+const value = someFn(someArg).throwIfUndefined()
+```
+
+### `TerriaErrorOverrides`
+
+`TerriaErrorOverrides` can be provided when creating a `new Result()`, and also when using `throwIfError()` or `throwIfUndefined`.
+This allows you to add more context to TerriaErrors.
+
+#### Simple usage
+
+```ts
+function someFn(someArg): Result<string | undefined> {
```

Comment:
It's not clear from this example which bit is the `TerriaErrorOverrides`. Add a comment?

Reply:
I added a comment